tests/test_validation.py

```diff
@@ -0,0 +1,162 @@
+
+
+def test_ModelObject_rules_access(
+    model: capellambse.MelodyModel,
+):
+    """Test convenient access of ModelObject rules with by_id,... ."""
+    obj = model.by_uuid(TEST_UUID)
+    assert isinstance(obj.validation, v.ElementValidation)
+
+    assert obj.validation.rules
+    assert (required_rules := obj.validation.rules["REQUIRED"])
+    assert required_rules.by_id(TEST_RULE_ID)  # type: ignore[attr-defined]
+    assert required_rules.by_category("REQUIRED")
+
+
+def test_MelodyModel_validation(model: capellambse.MelodyModel):
+    assert isinstance(model.validation, v.ModelValidation)
+    assert model.validation.rules
+
+    results = model.validate()
+
+    assert results
+
+
+@pytest.mark.parametrize(
+    "params", [pytest.param((TEST_RULE_ID, 1, 1, 13, 13), id=TEST_RULE_ID)]
+)
+def test_MelodyModel_validation_access(
+    model: capellambse.MelodyModel, params: tuple[str, int, int, int, int]
+):
+    rule_id, uuids, trues, categories, types = params
+    assert isinstance(model.validation, v.ModelValidation)
+    assert model.validation.rules
+
+    results = model.validate(rule=rule_id)
+
+    assert len(results.by_uuid(TEST_UUID)) == uuids
+    assert len(results.by_value(True)[rule_id]) == trues
+    assert len(results.by_category(v.Category.REQUIRED)[rule_id]) == categories
+    assert len(results.by_type("SystemComponent")[rule_id]) == types
+
+
+@pytest.mark.parametrize("type", [None, "SystemComponent"])
+def test_get_passed_and_total_results(
+    model: capellambse.MelodyModel, type: str | None
+):
+    assert isinstance(model.validation, v.ModelValidation)
+    assert model.validation.rules
+
+    results = model.validate()
+    passed, total = v.get_passed_and_total(results, type=type)
+
+    assert isinstance(passed, int) and isinstance(total, int)
+
+
+def test_ModelObject_validation(model: capellambse.MelodyModel):
+    obj = model.by_uuid(TEST_UUID)
+    assert isinstance(obj.validation, v.ElementValidation)
+    assert obj.validation.rules
+
+    results = obj.validate()
+
+    assert results
+
+
+@pytest.mark.parametrize("rule_id", [TEST_RULE_ID])
+def test_MelodyObject_validation_access(
+    model: capellambse.MelodyModel, rule_id: str
+):
+    obj = model.by_uuid(TEST_UUID)
+    assert isinstance(obj.validation, v.ElementValidation)
+    assert obj.validation.rules
+
+    results = obj.validate(rule=rule_id)
+
+    assert len(results) == 1
+    assert results[rule_id].uuid == obj.uuid
+
+
+# TODO: Test runtime for large models (50k+) to access results and rules
+
+
+def test_cli_writes_validation_report(tmp_path: pathlib.Path):
+    output_file = tmp_path / "report.html"
+    cli = subprocess.run(
+        [
+            sys.executable,
+            "-mcapellambse.extensions.validation",
+            "-m",
+            "coffee-machine",
+            "-o",
+            output_file,
+        ],
+        cwd=INSTALLED_PACKAGE.parent,
+        check=False,
+    )
```

Comment:
We should use [`click.testing`](https://click.palletsprojects.com/en/8.1.x/testing/) do implement proper tests, which can then also be measured for test coverage.